packages/material-ui/src/RadioGroup/RadioGroup.js

```diff
@@ -30,15 +39,22 @@ class RadioGroup extends React.Component {
     focusRadios[0].focus();
   };
 
-  handleRadioChange = (event, checked) => {
+  handleChange = (event, checked) => {
+    if (!this.isControlled) {
+      this.setState({
+        value: event.target.value,
```

Comment:
Shouldn't we only set this if `checked` is `true` and `null` it otherwise?

Reply:
I would go one step further by doing:
```diff
-    if (checked && this.props.onChange) {
+    if (this.props.onChange) {
      this.props.onChange(event, event.target.value);
    }
```

Reply:
`checked` should always be `true`, it's what I have empirically observed.

Reply:
I wasn't considering that `change` only fires on the selected radio element not the deselected one i.e. it's firing for a change of the value of a given radio group (by `name` attribute) not for a change of the `checked` attribute. Couldn't find explicit documentation about this. Though it is implied because `change` fires when the `value` attribute changes and not on any other attribute.

Reply:
@eps1lon Same observation as you. No documentation about, an empirical observation.

Reply:
Did some more digging and it does seem like no browser allows for deselecting a radio button. So for native browser behavior we are safe. It's still possible if it would be set to false and a change event would be fired imperatively. Just leaving this here for discoverability.

Reply:
@eps1lon It's good to know! I would vote for ignoring the imperative event. At least, until someone complains about it.